BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -4,34 +4,43 @@ import UIKit
 public struct AttributedLabel: Element {
 
     public var attributedText: NSAttributedString
-    public var numberOfLines: Int = 0
+    public var numberOfLines: Int
     /// The scale to which pixel measurements will be rounded. Defaults to `UIScreen.main.scale`.
     public var roundingScale: CGFloat = UIScreen.main.scale
 
-    public init(attributedText: NSAttributedString) {
+    public init(attributedText: NSAttributedString, numberOfLines: Int = 0) {
         self.attributedText = attributedText
+        self.numberOfLines = numberOfLines
     }
+    
+    /**
+     Keep around a label to use for measurement and sizing.
+     
+     We would usually do this using NSString or NSAttributedString's boundingRect family of methods,
+     but these do not let you specify a `numberOfLines` parameter, which is critical to correct sizing.
+     
+     As such, we will allocate this label once and then use it to measure by setting its text and attributes.
+     */
+    static let measurementLabel = UILabel()
```

Comment:
Verified that doing this does not have (m)any performance implications – measuring text 10,000 times with the label vs. the old boundingRect is about the same, +/- 5%.

Reply:
Oops, yep!